python/dgl/transform.py

```diff
@@ -575,6 +575,45 @@ def partition_graph_with_halo(g, node_part, num_hops):
         subg_dict[i] = subg
     return subg_dict
 
+def metis_partition(g, k, num_hops=1):
+    '''
+    This is to partition a graph with Metis partitioning.
+
+    The partitioned graph is stored in DGLGraph. The DGLGraph has the `inner_node` node data
+    that indicates whether a node is inside a partition and has the `inner_edge` edge data
+    that indicates whether an edge is inside a partition. In addition, it has the `part_id`
+    node data that indicates the partition a node belongs to.
+
+    Parameters
+    ------------
+    g: DGLGraph
+        The graph to be partitioned
+
+    k: int
+        The number of partitions.
+
+    num_hops: int
+        The number of hops a HALO node can be accessed.
+
+    Returns
+    --------
+    a dict of DGLGraphs
+        The key is the partition Id and the value is the DGLGraph of the partition.
+    '''
+    # METIS works only on symmetric graphs.
+    g = to_bidirected(g, readonly=True)
```

Comment:
Since this might be costly. Could you add a argument `is_bidirected=False`? So if the user already has a bidirectional graph, there is no need to perform conversion again.

Reply:
I benchmarked the one in GKLib. It's much faster. For a graph with (|V|=10000000, |E|=300000000), The GKLib implementation takes about 60 seconds with one thread. It doesn't look like it's a bottleneck.